### PR TITLE
[docs] Adding the guide on database initialization from tests (#2273)

### LINKED ISSUE #2273: INSERT/UPDATE operations are not executing for in-memory DB
**Description**
When using in-memory H2 database, data can be inserted or updated only on the schema initialization stage (via sql script). When INSERT/UPDATE executed in the story (via "When I execute SQL query..." step), updates to the data are not appearing.

**Steps To Reproduce**
1. Set up in-memory database for the project
2. Create a schema and table
3. Write scenario that contains insertion or update of the data in your database (please, see example in the Allure report attached)

**Expected behavior**
The data in in-memory database can be inserted/updates during runtime using SQL steps

**Artifacts**
[allure.zip](https://github.com/vividus-framework/vividus/files/7675472/allure.zip)

**Execution environment:**
 - Vividus version: 0.3.8
 - OS: Win 10
 - Java: 14.0.2

### PR DESCRIPTION
**Pull request by draker94**
Closes #2273

### CODE REVIEW

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 1]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
 implementation(group: 'com.h2database', name: 'h2', version: '2.0.202')
 ----
 
-. Configure connection to the H2 database
+. Configure connection to the H2 database and initialization
 +
+You could execute SQL on connection via the INIT property or through the step to execute SQL query.
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:20]:
```suggestion
DDL/DQL scripts could be run using either `INIT` property of the connection string or the step executing SQL queries.
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 2]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
...
+Whichever method of initializing a table is chosen, you could use RUNSCRIPT commands with the path to SQL files. Here you can specify absolute path to file or use the 'classpath:/' prefix - files will be searched from 'src/main/resources' folder relatively.
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:21]:
```suggestion
Whichever method of tables initialization is chosen, `RUNSCRIPT` statement with the path to SQL files could be used. Here you can specify absolute path to file or use the 'classpath:/' prefix - files will be searched from 'src/main/resources' folder relatively.
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 3]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
...
+[NOTE]
 The details on executing SQL statements on connection
 https://h2database.com/html/features.html#execute_sql_on_connection[could be found in the official H2 guide].
 +
-You could execute SQL on connection via the INIT property in two ways:
-
- - using RUNSCRIPT commands with the path to SQL files.
-Here you can specify absolute path to file
-or use the 'classpath:/' prefix - files will be searched from 'src/main/resources' folder relatively.
+.. Using INIT property
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:26]:
```suggestion
.. Using `INIT` property
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 4]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
...
+ - using INIT property data loaded into the database once and can no longer be change.
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:27]:
```suggestion
 - using `INIT` property data loaded into the database once and can no longer be change.
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 5]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
...
-  INSERT INTO students (firstname, lastname) VALUES ('Maria', 'Miller')\\;
+  RUNSCRIPT FROM 'classpath:/path/to/sql/file1.sql'\\;\
+  RUNSCRIPT FROM 'classpath:/path/to/sql/file2.sql'
 db.connection.testh2db.username=
 db.connection.testh2db.password=
 ----
-
-. Use the configured connection to execute SQL
+.. Using test step
+ - if you use table initialization in the step, you only need to specify connection parameters in the properties file and initialize the table in the test.
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:56]:
```suggestion
 - if you use tables initialization in the step, you only need to specify the connection parameters in the properties file and execute DDL script in the test.
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 6]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
...
+.Configure connection to database
+[source,properties]
+----
+db.connection.testh2db1.driver-class-name=org.h2.Driver
+db.connection.testh2db1.url=jdbc:h2:mem:test;\
+  DB_CLOSE_DELAY=-1;
+db.connection.testh2db1.username=
+db.connection.testh2db1.password=
+----
+WARNING: If you are using table initialization in the step, do not forget to set the property http://www.h2database.com/html/features.html#in_memory_databases[DB_CLOSE_DELAY].
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:67]:
```suggestion
WARNING: If you are using table initialization in the step, do not forget to set the property http://www.h2database.com/html/features.html#in_memory_databases[`DB_CLOSE_DELAY`].
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 7]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
...
++
+.Initialization database
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:69]:
```suggestion
.Initialization of the database
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 8]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
...
+----
+When I execute SQL query `
+  CREATE SCHEMA IF NOT EXISTS test;
+  SET SCHEMA test;
+  CREATE TABLE test.students (firstname VARCHAR(50) NOT NULL, lastname VARCHAR(50) NOT NULL);
+  INSERT INTO students (firstname, lastname) VALUES ('Maria', 'Miller');
+` against `testh2db1`
+----
++
+.Initialization database with RUNSCRIPT
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:80]:
```suggestion
.Initialization of database with `RUNSCRIPT`
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 9]
```diff
@@ -15,44 +15,79 @@ implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '{curre
...
+[source,gherkin]
+----
+When I execute SQL query `
+  CREATE SCHEMA IF NOT EXISTS test;
+  SET SCHEMA test;
+  RUNSCRIPT FROM 'classpath:/path/to/sql/file1.sql';
+  RUNSCRIPT FROM 'classpath:/path/to/sql/file2.sql'
+` against `testh2db1`
+----
+. Use the configured connection and the initialized database to execute SQL
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:90]:
```suggestion
----

. Use the configured connection and the initialized database to execute SQL
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 10]
```diff
@@ -61,3 +96,11 @@ When I execute SQL query `
 SELECT * FROM students
 ` against `testh2db` and save result to scenario variable `h2-records`
 ----
++
+.Execute SQL against H2 database if the table was created in the step with the schema 'test'
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:100]:
```suggestion
.Execute DML script against H2 database initialized in the step with the schema `test`
```

File: docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 11]
```diff
@@ -61,3 +96,11 @@ When I execute SQL query `
...
+[source,gherkin]
+----
+When I execute SQL query `
+SELECT * FROM test.students
```
Comment (valfirst) [inline docs/modules/guides/pages/use-in-memory-db-to-manage-test-data.adoc:104]: it makes sense to put `INSERT`, `UPDATE` or any other DML statement

File: vividus-tests/src/main/resources/properties/suite/integration/suite.properties [hunk 12]
```diff
@@ -43,3 +43,10 @@ db.connection.testh2db.url=jdbc:h2:mem:test;\
   INSERT INTO cosmetics (name, description) VALUES ('lipstick', 'Color with the power to transform your lips')\\;
 db.connection.testh2db.username=
 db.connection.testh2db.password=
+
+db.connection.testh2db1.driver-class-name=org.h2.Driver
+db.connection.testh2db1.url=jdbc:h2:mem:test;\
+  DB_CLOSE_DELAY=-1;\
+  DATABASE_TO_UPPER=FALSE;
+db.connection.testh2db1.username=
+db.connection.testh2db1.password=
```
Comment (valfirst) [inline vividus-tests/src/main/resources/properties/suite/integration/suite.properties:52]:
```suggestion
db.connection.testh2db-modifiable.driver-class-name=org.h2.Driver
db.connection.testh2db-modifiable.url=jdbc:h2:mem:test;\
  DB_CLOSE_DELAY=-1;\
  DATABASE_TO_UPPER=FALSE;
db.connection.testh2db-modifiable.username=
db.connection.testh2db-modifiable.password=
```

File: vividus-tests/src/main/resources/story/integration/Databases.story [hunk 13]
```diff
@@ -17,3 +17,18 @@ Then `<name>` is equal to `lipstick`
...
 Examples:
 {transformer=FROM_DB, dbKey=testh2db, sqlQuery=SELECT name\,description FROM cosmetics}
+
+Scenario: Verify execute sql steps without 'INIT' property and with 'DB_CLOSE_DELAY=-1' property in database connection url
+Meta:
+    @issueId 2273
+When I execute SQL query `
+CREATE SCHEMA if not exists test;
+SET SCHEMA test;
+CREATE TABLE if not exists test.launch_rockets (name VARCHAR(50) NOT NULL, 	payload_cost VARCHAR(100) NOT NULL, country VARCHAR(100) NOT NULL);
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/Databases.story:27]:
```suggestion
CREATE TABLE if not exists test.launch_rockets (name VARCHAR(50) NOT NULL, payload_cost VARCHAR(100) NOT NULL, country VARCHAR(100) NOT NULL);
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**